### PR TITLE
8295697: Resolve conflicts between serviceability/jvmti and nsk/jvmti shared code

### PR DESCRIPTION
**Pull request by alexmenkov**
The fix resolved conflicts between jvmti test shared code: test/lib/jdk/test/lib/jvmti/jvmti_common.h and test/hotspot/jtreg/vmTestbase/nsk/share/jvmti/* to allot NSK tests to use jvmti_common.h

testing: test native libraries are built before any test execution, so run build job should be enough, but for safety I also run test/hotspot/jtreg/vmTestbase/nsk/jvmti and test/hotspot/jtreg/serviceability/jvmti

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8295697](https://bugs.openjdk.org/browse/JDK-8295697): Resolve conflicts between serviceability/jvmti and nsk/jvmti shared code

### Reviewers
 * [Leonid Mesnik](https://openjdk.org/census#lmesnik) (@lmesnik - **Reviewer**)
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**)
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10733/head:pull/10733` \
`$ git checkout pull/10733`

Update a local copy of the PR: \
`$ git checkout pull/10733` \
`$ git pull https://git.openjdk.org/jdk pull/10733/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10733`

View PR using the GUI difftool: \
`$ git pr show -t 10733`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10733.diff">https://git.openjdk.org/jdk/pull/10733.diff</a>

</details>

### CODE REVIEW

**Review by lmesnik — CHANGES_REQUESTED** [review]
There is a lib ~/ws/jdk/open/test/lib/jdk/test/lib/jvmti/jvmti_common.h which has print_stack_trace. Could it be re-used instead?

File: test/hotspot/jtreg/vmTestbase/nsk/jvmti/PopFrame/popframe004/popframe004.cpp [hunk 1]
```diff
@@ -51,6 +51,71 @@ static void set_watch_ev(int value) {
...
+    printf("%02d: %s %s%s\n", depth,
+           className != NULL ? className : "<class name is not available>",
+           methodName != NULL ? methodName : "<method name is not available>",
+           methodSig != NULL ? methodSig : "<method signature is not available>");
+    jvmti->Deallocate((unsigned char*)classSig);
+    jvmti->Deallocate((unsigned char*)methodName);
+    jvmti->Deallocate((unsigned char*)methodSig);
+}
+
+static void printStackTrace(jvmtiEnv *jvmti, jthread thread) {
```
[the remark this answers is not in the document]
Comment (plummercj) [reply test/hotspot/jtreg/vmTestbase/nsk/jvmti/PopFrame/popframe004/popframe004.cpp:88]: Most of this code seems to have been copied from `Java_nsk_jvmti_GetStackTrace_getstacktr001_check`. Maybe some consolidating would be good.

**Comment by alexmenkov** [comment]
Thank you for review.
It was supposed to be a small local fix, but I agree with the point about code duplication.
We have 2 places with shared code for JVMTI tests:
test/lib/jdk/test/lib/jvmti/jvmti_common.h
and 
test/hotspot/jtreg/vmTestbase/nsk/share/jvmti/

jvmti_common.h contains both declaration and definitions, NSK shared files have separate .h and .cpp files (.cpp files are included in test .cpp files)

This 2 "libraries" have some duplicate functions and cannot be used simultaneously (a number of compiler errors).
I'm trying to resolve this without massive code change.
The plan is to include jvmti_common.h in some nsk shared files and remove duplicated functions from nsk files.
@lmesnik what do you think about the approach?

**Comment by lmesnik** [comment]
Sounds good to me. We could reuse print_frame after this.

**Comment by sspitsyn** [comment]
Nice plan, thanks!

**Comment by alexmenkov** [comment]
- resolved serviceability/jvmti and nsk/share/jvmti conflicts
- removed duplicated functions
- popframe004 test uses print_stack_trace from jvmti_common.h
- JVMTITools.h and JVMTITools.cpp are empty now, but the files are included in many nsk tests

File: test/hotspot/jtreg/vmTestbase/nsk/share/jni/jni_tools.h [hunk 2]
```diff
@@ -133,8 +131,9 @@ int nsk_jni_check_exception(JNIEnv* jni, const char file[], int line);
  * If value is negative, the first character of the stored string is
  * the minus sign (-). The function returns a pointer to the begining
  * of the result string.
+ * Declared and defined in jvmti_common.h
  */
-char *jlong_to_string(jlong value, char *string);
+//char *jlong_to_string(jlong value, char *string);
```
Comment (plummercj) [inline test/hotspot/jtreg/vmTestbase/nsk/share/jni/jni_tools.h:136]: It seems odd to have jlong_to_string and julong_to_string implemented in different places. Maybe more cleanup is needed here.

**Comment by plummercj** [comment]
I think the cleanup is disruptive enough that it should be a separate PR. Probably the easiest approach is to file a new CR for the cleanup and repurpose this PR for it, which means also moving the popframe004.cpp changes to a new PR.

File: test/hotspot/jtreg/vmTestbase/nsk/share/jni/jni_tools.h (hunk 2 above)
Comment (sspitsyn) [reply test/hotspot/jtreg/vmTestbase/nsk/share/jni/jni_tools.h:136]: I'd suggest to move julong_to_string to the same place where the jlong_to_string is placed and remove all these with the comments from jni_tools.

Comment (alexmenkov) [reply test/hotspot/jtreg/vmTestbase/nsk/share/jni/jni_tools.h]: I moved julong_to_string (and define for julong type) to jvmti_common.h
About JVMTITools.*
JVMTITools.cpp is included in 628 tests, JVMTITools.h is included in 348 files.
I don't think removing this 2 files worth such massive change

**Review by sspitsyn — APPROVED** [review]
Looks good to me.
Thank you for the update.
Serguei

**Review by lmesnik — APPROVED** [review]

**Pull request closed without merging**